extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/CustomQueryMethodsAdder.java

```diff
@@ -0,0 +1,271 @@
+                            throw new IllegalArgumentException(
+                                    method.name() + " of Repository " + repositoryClassInfo
+                                            + " is meant to be a delete query and can therefore only have a void or long return type");
+                        }
+
+                        // we need to strip 'delete' or else JpaOperations.delete will generate the wrong query
+                        String deleteQueryString = queryString.substring("delete".length());
+                        ResultHandle deleteCount;
+                        if (useNamedParams) {
+                            ResultHandle parameters = generateParametersObject(namedParameterToIndex, methodCreator);
+
+                            // call JpaOperations.delete
+                            deleteCount = methodCreator.invokeStaticMethod(
+                                    MethodDescriptor.ofMethod(JpaOperations.class, "delete", long.class,
+                                            Class.class, String.class, Parameters.class),
+                                    methodCreator.readInstanceField(entityClassFieldDescriptor, methodCreator.getThis()),
+                                    methodCreator.load(deleteQueryString), parameters);
+                        } else {
+                            ResultHandle paramsArray = generateParamsArray(queryParameterIndexes, methodCreator);
+
+                            // call JpaOperations.delete
+                            deleteCount = methodCreator.invokeStaticMethod(
+                                    MethodDescriptor.ofMethod(JpaOperations.class, "delete", long.class,
+                                            Class.class, String.class, Object[].class),
+                                    methodCreator.readInstanceField(entityClassFieldDescriptor, methodCreator.getThis()),
+                                    methodCreator.load(deleteQueryString), paramsArray);
+                        }
+
+                        if (DotNames.VOID.equals(methodReturnTypeDotName)) {
+                            methodCreator.returnValue(null);
+                        }
+                        handleLongReturnValue(methodCreator, deleteCount, methodReturnTypeDotName);
+
+                    } else if (queryString.startsWith("update") || queryString.startsWith("UPDATE")) {
+                        if (!DotNames.PRIMITIVE_INTEGER.equals(methodReturnTypeDotName)
+                                && !DotNames.INTEGER.equals(methodReturnTypeDotName)
+                                && !DotNames.VOID.equals(methodReturnTypeDotName)) {
+                            throw new IllegalArgumentException(
+                                    method.name() + " of Repository " + repositoryClassInfo
+                                            + " is meant to be an update query and can therefore only have a void or integer return type");
```

Comment:
Delete requires a long and update an integer? This is weird.

Reply:
Indeed. We could of course handle both if someone complains